src/store/facade/GroceryStore.java

```diff
@@ -479,6 +481,9 @@ public Result addProduct(Request request) {
 		Result result = new Result();
 		String productId = productsList.add(new Product(request.getProductName(), request.getProductId(),
 				request.getProductCurrentPrice(), request.getProductStockOnHand(), request.getProductReorderLevel()));
+
+		result.setProductFields(productsList.searchById(productId));
```

Comment:
This should be moved lower, inside the if clause (line 488); product fields should only be set if action was successful.

Reply:
Very good point,